homeassistant/components/surepetcare/__init__.py

```diff
@@ -110,22 +106,27 @@ async def async_setup(hass, config) -> bool:
         ]
     )
 
+    conf = config[DOMAIN]
+
+    # update interval
+    scan_interval = conf[CONF_SCAN_INTERVAL]
     # discover hubs the flaps/feeders are connected to
     hub_ids = set()
     for device in things.copy():
-        device_data = await surepy.device(device[CONF_ID])
+        device_data = await hass.async_add_executor_job(surepy.device, device[CONF_ID])
```

Comment:
The SurePetcare.device method is a coroutine function. It should not be scheduled on the executor. The problem is that the library is mixing asyncio with sync work, eg it both uses aiohttp and requests. This needs to be fixed in the library.